torch/_inductor/codegen/wrapper.py

```diff
@@ -212,6 +220,12 @@ def plan(self, state: MemoryPlanningState):
 
     def codegen(self, code: IndentedBuffer):
         assert self.node.get_name() not in V.graph.removed_buffers
+        for s in self.node.get_size():
+            s_str = str(s)
+            if s_str in V.graph.data_dep_symbol_to_expr_str:
+                expr_str = V.graph.data_dep_symbol_to_expr_str[s_str]
+                assert expr_str is not None
+                code.writeline(f"{s_str} = {expr_str}")
```

Comment:
I'm a little unsure whether or not this is the correct place to put this logic. When I was hacking through, I placed equivalent code in https://gist.github.com/ezyang/1293a41299604c44310341b7540eabcb#file-gistfile0-txt-L108-L121 What made you decide to put it here?

Reply:
cc @jansel for thoughts

Reply:
currently the generated code is roughly:
```python
i3 = buf0.sum().item()     # <----- this is the line of concern
buf2 = empty_strided((i3, s1), (s1, 1), device='cuda', dtype=torch.float32)
buf3 = [buf2]
buf3_work = dist.all_to_all_single(output=buf3[0], ...)
```
`buf2` which uses `i3` as dim size is input to `all_to_all_single`, so we might need to generate `i3` computation line here instead of in `MultiOutput`

Reply:
Removed the need for `data_dep_symbol_to_expr_str`